canoe/sp1-cc/client/src/main.rs

```diff
@@ -16,45 +15,31 @@ pub fn main() {
     let state_sketch = bincode::deserialize::<EvmSketchInput>(&state_sketch_bytes).unwrap();
 
     let verifier_address = sp1_zkvm::io::read::<Address>();
-    let batch_header_abi = sp1_zkvm::io::read::<Vec<u8>>();
-    let non_signer_stakes_and_signature_abi = sp1_zkvm::io::read::<Vec<u8>>();
-    let blob_inclusion_info_abi = sp1_zkvm::io::read::<Vec<u8>>();
-    let signed_quorum_numbers_abi = sp1_zkvm::io::read::<Vec<u8>>();
+    let canoe_input = sp1_zkvm::io::read::<CanoeInput>();
 
     // Initialize the client executor with the state sketch.
     // This step also validates all of the storage against state root provided by the host
     let executor = ClientExecutor::new(&state_sketch).unwrap();
 
-    // Execute the slot0 call using the client executor.
-    let batch_header = <BatchHeaderV2 as SolType>::abi_decode(&batch_header_abi)
-        .expect("deserialize BatchHeaderV2");
-    let blob_inclusion_info = <BlobInclusionInfo as SolType>::abi_decode(&blob_inclusion_info_abi)
-        .expect("deserialize BlobInclusionInfo");
-    let non_signer_stakes_and_signature =
-        <NonSignerStakesAndSignature as SolType>::abi_decode(&non_signer_stakes_and_signature_abi)
-            .expect("deserialize NonSignerStakesAndSignature");
-
-    let signed_quorum_numbers = Bytes::abi_decode(&signed_quorum_numbers_abi)
-        .expect("deserialize signed_quorum_numbers_abi");
-
-    let mock_call = IEigenDACertVerifier::verifyDACertV2ForZKProofCall {
-        batchHeader: batch_header,
-        blobInclusionInfo: blob_inclusion_info.clone(),
-        nonSignerStakesAndSignature: non_signer_stakes_and_signature,
-        signedQuorumNumbers: signed_quorum_numbers,
+    // TODO, are there no better way to reduce this replicated code.
+    // known constraint, new_call takes SolCall trait, which is Sized so not dyn trait
+    // also impl SolCall seems too much
```

Comment:
not sure what you are talking about here, is this comment really needed or is it stream of consciousness?

Reply:
what does "seems too much" mean?